### PR TITLE
Added parameters to response(). Added 'format' parameter to as.respon…

### PR DESCRIPTION
**Pull request by pteetor**
Added parameters to response(). Added 'format' parameter to as.response.data.frame(). Probably made other changes, too.

The changes are intended to simplify implementation of the microservice.

### CODE REVIEW

File: R/response.R [hunk 1]
```diff
@@ -148,10 +152,29 @@ as.response.character <- function(x, directory = 'views', collapse = '\n', ...)
...
-as.response.data.frame <- function(x, ...) {
-  res <- response()
-  body(res) <- as.json(x, ...)
-  res
+as.response.data.frame <- function(x, format="json", ...) {
+  switch(format,
+         json = response(content_type = "application/json",
+                         body = as.json(x, ...) ),
+         html = response(content_type = "text/html",
+                         body = as.character(htmlTable::htmlTable(x)) ),
```
Comment (nteetor) [inline R/response.R]: I would like to allow an HTML format (seems standard), however I am averse to adding another package dependency. 

(I am looking to remove the `stringr` dependency soon.)

File: R/response.R [hunk 2]
```diff
@@ -148,10 +152,29 @@ as.response.character <- function(x, directory = 'views', collapse = '\n', ...)
...
+         csv = {
+          theText = NULL
```
Comment (nteetor) [inline R/response.R]: (snake case)

File: R/response.R [hunk 3]
```diff
@@ -73,14 +77,14 @@
 #' mkup('get', '302')
 #' mkup('get', '203')
 #'
-response <- function() {
+response <- function(status=200, content_type='text/plain', body='') {
```
Comment (nteetor) [inline R/response.R:80]: Small formatting change, "=" to " = ".

Comment (nteetor) [inline R/response.R:80]: Do you think adding a `headers` parameter, a list of values, is too open ended?

File: R/response.R (hunk 2 above)
Comment (nteetor) [inline R/response.R]: Are "="s required inside a switch statement or can "<-"s be used instead?

File: R/response.R (hunk 3 above)
Comment (nteetor) [reply R/response.R:80]: (there are a few more "=" instances further down too)

**Review by nteetor — APPROVED** [review]
Don't forget spaces around equals signs.

**Comment by pteetor** [comment]
Thank you for editing and accepting these changes! I'll try to track your style more carefully in the future.

File: R/response.R (hunk 2 above)
Comment (pteetor) [reply R/response.R]: The switch "statement" is really a function call, not a syntactic structure. So it must conform to function-call syntax, and the "<-" syntax would not work.

File: R/response.R (hunk 3 above)
Comment (pteetor) [reply R/response.R:80]: I think `headers` could be very useful. It the need arises in my work, I'll add it.

**Pull request merged**